### DISCUSSION TITLE
MOOSE Build Error

### DISCUSSION THREAD

**Discussion by aarograh** [idaholab/moose#19604] [Q&A Getting Started] [answered]
I'm trying get get a MOOSE environment set up for myself.  I'm working in the windows subsystem for linux, so it's ubuntu 20.04.  I had an old environment and tried updating it, but I hit the build error below.  I decided to just completely wipe my MOOSE repo and miniconda installation to start from scratch.  I did that, then exactly followed the instructions here: https://mooseframework.inl.gov/getting_started/installation/conda.html .  I encountered no errors along the way.  However, upon trying to build, I'm still getting the exact same sets of error messages shown below.  Any ideas what's going on?  It seems like it's looking for libmesh in the moose directory instead of in the conda environment?

```
(moose) aarograh@PC0123449:~/ag6-windows/repos/moose/test$ make -j 4
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/libtool: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/libtool: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
Using HIT from /home/aarograh/ag6-windows/repos/moose/framework/contrib/hit
Rebuilding symlinks in /home/aarograh/ag6-windows/repos/moose/framework/build/header_symlinks
Rebuilding symlinks in /home/aarograh/ag6-windows/repos/moose/test/build/header_symlinks
Checking if header needs updating: /home/aarograh/ag6-windows/repos/moose/framework/include/base/MooseRevision.h...
Creating Unity Directory /home/aarograh/ag6-windows/repos/moose/framework/build/unity_src
Copying default MOOSE configuration to: /home/aarograh/ag6-windows/repos/moose/framework/include/base/MooseConfig.h...
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
Compiling C++ (in opt mode) /home/aarograh/ag6-windows/repos/moose/framework/contrib/pcre/src/pcrecpp.cc...
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/libtool: not found
Compiling C++ (in opt mode) /home/aarograh/ag6-windows/repos/moose/framework/contrib/pcre/src/pcre_stringpiece.cc...
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/contrib/bin/libmesh-config: not found
make: *** [/home/aarograh/ag6-windows/repos/moose/framework/build.mk:119: /home/aarograh/ag6-windows/repos/moose/framework/contrib/pcre/src/pcrecpp..opt.lo] Error 127
make: *** Waiting for unfinished jobs....
/bin/sh: 1: /home/aarograh/ag6-windows/repos/moose/libmesh/installed/libtool: not found

```

**Comment by dschwen** [comment, marked as answer]
Try closing your terminal and opening it again. You command line prompt should contain `(moose)` now. If not then you didn't follow the instructions _exactly_ :-). You might be missing the conda init step or the conda activate moose step.

**Reply by aarograh** [reply]
I _did_ follow the instructions exactly and my terminal contained `(moose)` like you said (I've done this before, it has just been awhile).  

That being said, I closed the terminal and tried again and it seems to be working.  Re-reading the instructions, I think this part may have been the source of confusion:

>
"Where SHELL_NAME is the shell you discovered in the previous step. Once complete, close any terminals you have opened, and re-open them. Then run conda activate moose again. If successful proceed to installing packages:"
>

That sounded to me like you only have to reopen the terminals if you encountered errors with your shell, but in light of your comment it seems that you always need to restart the terminal.  I'd recommend rewording that to be a bit more clear if it is expected that the user always needs to reopen their terminal.

Thanks for the suggestion.

**Reply by milljm** [reply]
We totally agree, and are working on those instructions currently: https://github.com/idaholab/moose/pull/19453

The mock site is available here: https://mooseframework.inl.gov/docs/PRs/19453/site/getting_started/installation/conda.html

I believe if these instructions were instead made available, you would have had a better experience.

**Reply by aarograh** [reply]
Jason, thanks, good to know.   That does look a little more clear, and it's nice that the references to conda are being changed to mamba.  I see a few lingering commands that use `conda` still (adding channels and the uninstall instructions)... have those not been updated yet, or is conda actually the correct command in those instances?

**Reply by milljm** [reply]
Mamba has yet to "wrap" certain commands unfortunately. So the ones you see still stating `conda` are necessary :( I am ... like 99% sure we checked them all!

I think anything relating to a configuration will probably have to remain `conda` (aside from `mamba init`), like adding channels and the like. We don't like it... we would love to see 100% mamba. Its faster, would be less confusing, and seem to just "do" everything better (like auto-discovering your shell).

**Reply by aarograh** [reply]
Yes, that's annoying, but obviously outside of your control.  Maybe something will get updated with conda/mamba to clean that up in the near future.  Thank you for the clarifications!
